### PR TITLE
finish

### PR DESCRIPTION
**Pull request by ninahur1989**

### CODE REVIEW

File: GRUD/Get.cs [hunk 1]
```diff
@@ -0,0 +1,69 @@
+﻿namespace Networking
+{
+    using Networking.Inerface;
+    public class Get: IChecker
```
Comment (vladboss61) [inline GRUD/Get.cs]: make class sealed, and add space like this Get : IChecker

File: GRUD/Get.cs [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+                Console.WriteLine($"\nsuccess {numOfTask} TASK ");
+                string content = await res.Content.ReadAsStringAsync();
+                Console.WriteLine(content);
+            }
+        }
+
+        public async Task GetAsync()
+        {
+            using var client = new HttpClient();        
+            client.BaseAddress = new Uri("https://reqres.in/");
```
Comment (vladboss61) [inline GRUD/Get.cs]: add constant "https://reqres.in/" in new file and reuse it everywhere in the project

File: GRUD/Post.cs [hunk 3]
```diff
@@ -0,0 +1,111 @@
...
+    using Networking.Models;
+    using Newtonsoft.Json;
+    using System.Text;
+
+    internal class Post : IChecker
+    {
+        public async Task PostAsync()
+        {
+            using var client = new HttpClient();
+            client.BaseAddress = new Uri("https://reqres.in/");
```
Comment (vladboss61) [inline GRUD/Post.cs]: Reuse constant

File: Inerface/IChecker.cs [hunk 4]
```diff
@@ -0,0 +1,12 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Networking.Inerface
+{
+    internal interface IChecker
```
Comment (vladboss61) [inline Inerface/IChecker.cs]: remove useless empty interface or rename it something like it `IReqres` with method `public void ProccessRequest()` and implement two of your classes with that method.

File: GRUD/Post.cs [hunk 5]
```diff
@@ -0,0 +1,111 @@
+﻿namespace Networking.GRUD
+{
+    using Networking.Inerface;
+    using Networking.Models;
+    using Newtonsoft.Json;
+    using System.Text;
+
+    internal class Post : IChecker
```
Comment (vladboss61) [inline GRUD/Post.cs]: make sealed

File: Program.cs [hunk 6]
```diff
@@ -1,2 +1,11 @@
-﻿// See https://aka.ms/new-console-template for more information
-Console.WriteLine("Hello, World!");
+﻿namespace Networking 
+{
+    internal class Program
+    {
+        static void Main(string[] args)
```
Comment (vladboss61) [inline Program.cs]: make `private or internal` as you want

File: Program.cs [hunk 7]
```diff
@@ -1,2 +1,11 @@
-﻿// See https://aka.ms/new-console-template for more information
-Console.WriteLine("Hello, World!");
+﻿namespace Networking 
+{
+    internal class Program
```
Comment (vladboss61) [inline Program.cs]: add sealed.

File: Starter.cs [hunk 8]
```diff
@@ -0,0 +1,17 @@
+﻿namespace Networking
+{
+    using System.Net;
+    using System.Text;
+    using Networking.GRUD;
+    using Newtonsoft.Json;
+
+    internal class Starter
```
Comment (vladboss61) [inline Starter.cs]: add sealed

File: Models/RegisterUserParameters.cs [hunk 9]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Networking.Models
+{
+    using Newtonsoft.Json;
```
Comment (vladboss61) [inline Models/RegisterUserParameters.cs:3]: add enter to the next line

File: GRUD/Post.cs [hunk 10]
```diff
@@ -0,0 +1,111 @@
...
+            await RegisterAsync(userlog1, client, "12", "api/register");
+            //13
+            await RegisterAsync(userlog2, client, "13", "api/register");
+            //14
+            await RegisterAsync(userlog1, client, "14", "api/login");
+            //15
+            await RegisterAsync(userlog2, client, "15", "api/login");
+        }
+
+        public async Task StatusCheckerAsync(HttpResponseMessage res, string numOfTask, string message)
```
Comment (vladboss61) [inline GRUD/Post.cs]: rename res  -> request, rename `StatusCheckerAsync -> DisplaySuccessfullResponse`

File: Models/User.cs [hunk 11]
```diff
@@ -0,0 +1,12 @@
+﻿namespace Networking
+{
+    public sealed class User
+    {
+        public string Name { get; set; }
+
+        public string Job { get; set; }
+
+        public int? Id { get; set; }
```
Comment (vladboss61) [inline Models/User.cs:9]: add enter in the end

File: Models/UserToken.cs [hunk 12]
```diff
@@ -0,0 +1,8 @@
+﻿namespace Networking.Models
+{
+    public sealed class UserToken
+    {
+        public int Id { get; set; }
```
Comment (vladboss61) [inline Models/UserToken.cs]: add enter in the end

Comment (vladboss61) [inline Models/UserToken.cs]: make field nullable

File: GRUD/Post.cs [hunk 13]
```diff
@@ -0,0 +1,111 @@
...
+
+            HttpResponseMessage result5 = await client.PostAsync(path, stringContent3);
+            if (result5.IsSuccessStatusCode)
+            {
+                Console.WriteLine($"\nTask {task} is successed using POST({path})");
+                string content = await result5.Content.ReadAsStringAsync();
+
+                UserToken userToken = JsonConvert.DeserializeObject<UserToken>(content);
+                Console.WriteLine(content);
+            }
```
Comment (vladboss61) [inline GRUD/Post.cs:106]: remove else, please take out the content of else block above scope.

File: GRUD/Post.cs [hunk 14]
```diff
@@ -0,0 +1,111 @@
...
+            StringContent stringContent3 = new StringContent(serializedUser3, Encoding.Unicode, "application/json");
+
+            HttpResponseMessage result5 = await client.PostAsync(path, stringContent3);
+            if (result5.IsSuccessStatusCode)
+            {
+                Console.WriteLine($"\nTask {task} is successed using POST({path})");
+                string content = await result5.Content.ReadAsStringAsync();
+
+                UserToken userToken = JsonConvert.DeserializeObject<UserToken>(content);
+                Console.WriteLine(content);
```
Comment (vladboss61) [inline GRUD/Post.cs:105]: add `return Task.CompletedTask;` in the end of if block

File: GRUD/Get.cs (hunk 2 above)
Comment (ninahur1989) [reply GRUD/Get.cs]: +

**Pull request closed without merging**